source/acquaman/SGM/SGMHexapodMapScanActionControllerAssembler.cpp

```diff
@@ -0,0 +1,207 @@
+#include "SGMHexapodMapScanActionControllerAssembler.h"
+
+#include "actions3/AMListAction3.h"
+#include "actions3/AMActionSupport.h"
+#include "actions3/actions/AMAxisStartedAction.h"
+#include "actions3/actions/AMAxisFinishedAction.h"
+#include "actions3/actions/AMAxisValueFinishedAction.h"
+#include "actions3/actions/AMDetectorTriggerAction.h"
+#include "beamline/AMDetectorTriggerSource.h"
+
+#include "beamline/CLS/CLSAMDSScalerChannelDetector.h"
+#include "beamline/SGM/SGMBeamline.h"
+#include "beamline/SGM/SGMHexapodTransformedAxis.h"
+
+#include <QDebug>
+#include <QVector3D>
+
+SGMHexapodMapScanActionControllerAssembler::SGMHexapodMapScanActionControllerAssembler(QObject *parent)
+	: AMScanActionControllerScanAssembler(parent)
+{
+
+}
+
+bool SGMHexapodMapScanActionControllerAssembler::generateActionTreeImplmentation()
+{
+	if (axes_.size() != 2)
+		return false;
+
+	// Grab the axes individually for conceptual ease later.
+	AMScanAxis *stepAxis = axes_.at(0);
+	AMScanAxis *continuousAxis = axes_.at(1);
+
+	// Grab the controls individually for conceptual ease later.
+	AMControl *stepControl = controls_->at(0);
+	AMControl *continuousControl = controls_->at(1);
+
+	// First we need to make the list for the first axis.
+	AMListAction3 *mapActions = new AMListAction3(new AMListActionInfo3(QString("Axis %1").arg(stepAxis->name()), QString("Axis %1").arg(stepAxis->name())), AMListAction3::Sequential);
+
+	AMAxisStartedAction *axisStartAction = new AMAxisStartedAction(new AMAxisStartedActionInfo(QString("%1 Axis").arg(stepAxis->name()), AMScanAxis::StepAxis));
+	mapActions->addSubAction(axisStartAction);
+
+	// Get the initialization actions for the step axis.
+	AMListAction3 *initializationActions = new AMListAction3(new AMListActionInfo3(QString("Initializing %1").arg(stepControl->name()), QString("Initializing Axis with Control %1").arg(stepControl->name())), AMListAction3::Sequential);
+
+	AMAction3 *initializeControlPosition = AMActionSupport::buildControlMoveAction(stepControl, stepAxis->axisStart());
+	initializeControlPosition->setGenerateScanActionMessage(true);
+	initializationActions->addSubAction(initializeControlPosition);
+	mapActions->addSubAction(initializationActions);
+
+	// There will be only 1 region in these step scans because it is a map.
+	AMScanAxisRegion *stepRegion = stepAxis->regionAt(0);
+	QList<double> stepRegionPositions = generateStepAxisPositions(stepRegion);
+	// Grabbing current position so we know what the z value is.
+	QVector3D currentPosition = QVector3D(SGMBeamline::sgm()->exposedControlByName("Hexapod Global X Axis")->setpoint(),
```

Comment:
I think this is also going to run into trouble. From looking at it, it seems that it is going to perform the requested scan in the native coordinate system of the hexapod, not the transformed axes.